go-controller/cmd/ovnkube-trace/ovnkube-trace.go

```diff
@@ -160,251 +225,261 @@ func getPodMAC(client *corev1client.CoreV1Client, pod *kapi.Pod) (podMAC string,
 	return podMAC, nil
 }
 
-func getSvcInfo(coreclient *corev1client.CoreV1Client, restconfig *rest.Config, svcName string, ovnNamespace string, namespace string, cmd string) (svcInfo *SvcInfo, err error) {
+// getOvnKubePodOnNode returns the name of the ovnkube-node pod that is running on a given node.
+func getOvnKubePodOnNode(coreclient *corev1client.CoreV1Client, ovnNamespace string, nodeName string) (string, error) {
+	// Get pods in the openshift-ovn-kubernetes namespace
+	podsOvn, errOvn := coreclient.Pods(ovnNamespace).List(context.TODO(), metav1.ListOptions{})
+	if errOvn != nil {
+		klog.V(0).Infof("Cannot find pods in %s namespace, err: %v", ovnNamespace, errOvn)
+		return "", errOvn
+	}
+	var ovnkubePod *kapi.Pod
+	// Find ovnkube-node-xxx pod running on the same node as Pod
+	for _, podOvn := range podsOvn.Items {
+		podOvn := podOvn
+		if podOvn.Spec.NodeName == nodeName {
+			if !strings.HasPrefix(podOvn.Name, "ovnkube-node-metrics") {
+				if strings.HasPrefix(podOvn.Name, "ovnkube-node") {
+					klog.V(5).Infof("==> pod %s is running on node %s", podOvn.Name, nodeName)
+					ovnkubePod = &podOvn
+					break
+				}
+			}
+		}
+	}
+	if ovnkubePod == nil {
+		err := fmt.Errorf("cannot find ovnkube-node pod on node %s in namespace %s", nodeName, ovnNamespace)
+		return "", err
+	}
+	return ovnkubePod.Name, nil
+}
+
+// findPodOvsInterfaceInformation searches the node's OVS database for information about this pod's OVS interface.
+func findPodOvsInterfaceInformation(coreclient *corev1client.CoreV1Client, restconfig *rest.Config, ovnNamespace, ovnkubePodName, fullyQualifiedPodName string) (map[string]string, error) {
+	interfaceInfo := make(map[string]string)
 
+	findInterfaceCmd := fmt.Sprintf("ovs-vsctl find interface external_ids:iface-id=%s", fullyQualifiedPodName)
```

Comment:
I just realised, the older command is way simpler! 
```
"ovs-vsctl get Interface " + podInfo.VethName + " ofport"
```

Reply:
I need both the interface name and ofport, though, so that I can get the vethname:
~~~
   469         ovsInterfaceInformation, err := getPodOvsInterfaceInformation(coreclient, restconfig, ovnNamespace, podInfo.OvnKubePodName, podInfo.FullyQualifiedPodName())
   470         if err != nil {
   471             return nil, err
   472         }
   473         name, ok := ovsInterfaceInformation["name"]
   474         if !ok {
   475             return nil, fmt.Errorf("interface name not found in interface information: %v", ovsInterfaceInformation)
   476         }
   477         ofport, ok := ovsInterfaceInformation["ofport"]
   478         if !ok {
   479             return nil, fmt.Errorf("interface ofport not found in interface information: %v", ovsInterfaceInformation)
   480         }
   481         podInfo.PrimaryInterfaceName = "eth0"
   482         podInfo.VethName = strings.Trim(name, "\"")
   483         podInfo.OfportNum = ofport
~~~

That way, I can fix: https://bugzilla.redhat.com/show_bug.cgi?id=2066891#c2

I will reduce it to those 2 fields (name and ofport). I will also factor this part out into its own commit.